Sources/Apodini/Application/Application+Exporters.swift

```diff
@@ -59,7 +59,8 @@ extension Application {
             return self.application.storage[Key.self]!
         }
 
-        var semanticModelBuilderBuilder: (SemanticModelBuilder) -> (SemanticModelBuilder) {
+        /// The `SemanticModelBuilder` to register services upon
+        public var semanticModelBuilderBuilder: (SemanticModelBuilder) -> (SemanticModelBuilder) {
```

Comment:
To be honest I think that the current handling of the SemanticModelBuilder as well as how it is extended in this PR is very complicated and making it public is also an addition to the API surface. As far as I can see the reason for making the SemanticModelBuilder public is to easily register the interface exporters in the configure method.

I would suggest to following which doesn't require the SemanticModelBuilder to be public as this is really a class that a normal user shouldn't really need to care about. I would suggest the following:
Add a computed property to `Application` that stores a collection of `AnyInterfaceExporter` and maybe add convenience methods to application such as `addInterfaceExporter(...)` for `InterfaceExporter` and `StaticInterfaceExporter`. This way the interface exporters are collected in the application, the SemanticModelBuilder doesn't need to be passed around and the `SemanticModelBuilder` can be created after all configurations have been run and gets the interface exporter collection passed in in the initializer or accesses it using the `Application`:
```swift
extension Application {
    private final class InterfaceExporterStorage {
        var exporters: [AnyInterfaceExporter] = []
        
        func append(_ exporter: AnyInterfaceExporter) {
            exporters.append(exporter)
        }
    }
    
    private enum InterfaceExporerKey: StorageKey {
        typealias Value = InterfaceExporterStorage
    }
    
    var interfaceExporters: [AnyInterfaceExporter] {
        get {
            storage[InterfaceExporerKey.self]?.exporters ?? []
        }
    }
    
    private var interfaceExporterStorage: InterfaceExporterStorage {
        get {
            guard let exporterStorage = storage[InterfaceExporerKey.self] else {
                let exporterStorage = InterfaceExporterStorage()
                storage[InterfaceExporerKey.self] = exporterStorage
                return exporterStorage
            }
            return exporterStorage
        }
    }
    
    /// Registers an `InterfaceExporter` instance on the model builder.
    /// - Parameter instance: The instance to register.
    /// - Returns: `Self`
    public func register<T: InterfaceExporter>(exporter instance: T) {
        interfaceExporterStorage.append(AnyInterfaceExporter(instance))
    }
    
    /// Registers an `StaticInterfaceExporter` instance on the model builder.
    /// - Parameter instance: The instance to register.
    /// - Returns: `Self`
    public func register<T: StaticInterfaceExporter>(exporter instance: T) {
        interfaceExporterStorage.append(AnyInterfaceExporter(instance))
    }
}
```

The `SemanticModelBuilder` drops the with methods and just gets the interface exporters in the initializer.

The `start` method changes to something like this:
```swift
    static func start(app: Application, webService: Self = Self()) {
        /// Configure application and instanciate exporters
        webService.configuration.configure(app)
        
        /// If no specific address hostname is provided we bind to the default address to automatically and correcly bind in Docker containers.
        if app.http.address == nil {
            app.http.address = .hostname(HTTPConfiguration.Defaults.hostname, port: HTTPConfiguration.Defaults.port)
        }
        
        webService.register(
            SemanticModelBuilder(app, interfaceExporters: app.interfaceExporters)
        )
    }
```

This can then also remove the logic in this file here.

Reply:
Maybe it would even make sense to merge the passed `InterfaceExporters` in the initializer of `SemanticModelBuilder` with the `InterfaceExporters` in Application.interfaceExporters? Of course, we would need to check for duplicates (eg. not two RESTExporters, even if they have different configurations) and prioritize the passed `InterfaceExporters` over Application.interfaceExporters

Reply:
I see, I missed that you already pass the app instance in the `SemanticModelBuilder`. Just calling `SemanticModelBuilder(app)` is sufficient and you can also remove the interface builder related property and functions in the `SemanticModelBuilder`, this is then just duplicated and everything should be stored in the `app` instance and should be retrieved from there.